src/server/scripts/EasternKingdoms/zone_silverpine_forest.cpp

```diff
@@ -1596,6 +1596,201 @@ struct npc_silverpine_forsaken_bat : public VehicleAI
+        }
+        else
+            _events.ScheduleEvent(EVENT_SNIFFING, 2s, 12s);
+    }
+
+    void JustEngagedWith(Unit* who) override
+    {
+        me->RemoveAura(SPELL_STALKING);
+        me->RemoveAura(SPELL_KILL_ME_QUEST);
+
+        _events.CancelEvent(EVENT_SNIFFING);
+        _events.CancelEvent(EVENT_STRIKE_BOAR);
+
+        _events.ScheduleEvent(EVENT_SHADOWSTEP, 250ms);
+        _events.ScheduleEvent(EVENT_SINISTER_STRIKE, 6s, 8s);
+    }
+
+    void JustDied(Unit* killer) override
+    {
+        if (Player* player = killer->ToPlayer())
+        {
+            if (player->GetQuestStatus(QUEST_EXCISING_THE_TAINT) == QUEST_STATUS_NONE)
+            {
+                if (const Quest* ExcisingTheTaint = sObjectMgr->GetQuestTemplate(QUEST_EXCISING_THE_TAINT))
+                    player->AddQuest(ExcisingTheTaint, nullptr);
+            }
+        }
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_SNIFFING:
+                    me->HandleEmoteCommand(EMOTE_ONESHOT_SNIFF);
+                    _events.Repeat(18s, 25s);
+                    break;
+
+                case EVENT_STRIKE_BOAR:
+                    if (Creature* boar = me->FindNearestCreature(NPC_DARKTUSK_BOAR, 5.0f))
+                        me->CastSpell(boar, SPELL_BLOOD_STRIKE, true);
+                    _events.Repeat(5s, 7s);
+                    break;
+
+                case EVENT_SHADOWSTEP:
+                    DoCastVictim(SPELL_SHADOWSTEP);
+                    _events.Repeat(14s, 15s);
+                    break;
+
+                case EVENT_SINISTER_STRIKE:
+                    DoCastVictim(SPELL_SINISTER_STRIKE);
+                    _events.Repeat(6s, 8s);
+                    break;
+
+                default:
+                    break;
+            }
+        }
+
+        if (!UpdateVictim())
+            return;
+
+        DoMeleeAttackIfReady();
+    }
+
+private:
+    EventMap _events;
+};
+
+enum CaretakerSmithers
+{
+    SPELL_FACE_RIP                              = 84440,
+    SPELL_THROW_LANTERN                         = 81764,
+    SPELL_FRENZY                                = 81173,
+
+    EVENT_SMITHERS_THROW_LANTERN                = 1,
+    EVENT_SMITHERS_FRENZY                       = 2,
+
+    TALK_SMITHERS_AGROO                         = 0,
+    TALK_SMITHERS_LANTERN                       = 1,
+    TALK_SMITHERS_FRENZY                        = 2
+};
+
+// 45219 - Caretaker Smithers
+struct npc_silverpine_caretaker_smithers : public ScriptedAI
+{
+    npc_silverpine_caretaker_smithers(Creature* creature) : ScriptedAI(creature) { }
+
+    void JustAppeared() override
+    {
+        _homePosition = me->GetPosition();
+    }
+
+    void Reset() override
+    {
+        _events.Reset();
+
+        me->NearTeleportTo(_homePosition, false);
```

Comment:
with `me->GetHomePosition()` you could drop `_homePosition`

Reply:
Feedback applied.